components/library/src/pagination/mod.rs

```diff
@@ -45,11 +45,13 @@ impl<'a> Pager<'a> {
 #[derive(Clone, Debug, PartialEq)]
 pub struct Paginator<'a> {
     /// All pages in the section/taxonomy
-    all_pages: &'a [DefaultKey],
+    all_pages: Vec<DefaultKey>,
```

Comment:
Would `Cow` work here to avoid the performance penalty when cloning the keys if we don't need to?

Reply:
good suggestion, I added it in the next commit